src/vm/jitinterface.cpp

```diff
@@ -6640,6 +6643,61 @@ const char* CEEInfo::getMethodNameFromMetadata(CORINFO_METHOD_HANDLE ftnHnd, con
     return result;
 }
 
+/*********************************************************************/
+const char* CEEInfo::getClassNameFromMetadata(CORINFO_CLASS_HANDLE cls, const char** namespaceName)
+{
+    CONTRACTL {
+        SO_TOLERANT;
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    const char* result = NULL;
+    const char* namespaceResult = NULL;
+
+    JIT_TO_EE_TRANSITION();
+    TypeHandle VMClsHnd(cls);
+    result = VMClsHnd.GetMethodTable()->GetFullyQualifiedNameInfo(&namespaceResult);
```

Comment:
`VMClsHnd.GetMethodTable()` can return null for special type handles.

To avoid this issue, it may be better to make this for TypeHandles that are not TypeDescs only:

```
if (!VMClsHnd.IsTypeDesc())
{
     result = VMClsHnd.AsMethodTable()->GetFullyQualifiedNameInfo(&namespaceResult);
...
```

Reply:
Thank you! Will fix all the issue soon.